source/index.html.md

```diff
@@ -630,6 +630,27 @@ object][ApplePay-PaymentToken] for more information.
 </p>
 
 
+##### Method: `samsungpay`
+
+Samsung Pay provides the payment details as either JWE or JWE/JWS JOSE objects.
+Currently only the JWE is accepted.
+
+<dl class="dl-vertical">
+  <dt>samsungpay[jwt]</dt>
```

Comment:
I can't help 👃 the name ... sorry 🙏 ... that is, I still dislike that we don't have a *name* for the parameter but us the "format" as the parameter name.

Would we want to accept "JWE tokens" and "JWE/JWS tokens" in the same parameter when/if we support both? With `[3dsecure][v2]` we ask to get `ares` and `rreq` in two different parameters — perhaps this was a design flaw?

In Samsung Pay - Web checkout - Integration guide, the name "payment credential data" and just "credential data" is used. Perhaps `samsungpay[credential_data]` would make sense?

Reply:
You don't have to be sorry, we're being pedantic about docs, that's fine :)

Let me just state my preferences:

- If we decide to accept both JWE and JWS as the same parameter, I prefer `jwt`
- If we decide to accept the separately, I prefer `jwe` and `jws`.

It's common around the web to use JWE and JWS as nouns. We could also use `jwt_string`, but that's uglier to me. (The guide you referred to and the other guides also use JWE as a noun)

When I read `samsungpay[jwt]` I read it as _"the Samsung pay JWT"_, which makes perfect sense, as the JWT is in a context here.


However _"the Samsung pay credential data"_ doesn't immediately make sense, is it the unpacked JWT payload? The only thing it has going for it is that it's used in the web checkout guide.

But I don't want to bikeshed this anymore, you're free to select whichever you like :)


Now, if we choose to also accept the JWS, then we should likely follow the 3DSv2 precedence and split them to accept either the JWS or the JWT.
(Though admittedly, I would strongly prefer them to be a single parameter)